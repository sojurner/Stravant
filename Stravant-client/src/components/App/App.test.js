// import React from 'react';
// import { mapStateToProps, mapDispatchToProps, App } from './App';
// import * as store from '../../mockData/mockStore';
// import { setAccessToken } from '../../actions/userAction';

// it('renders without crashing', done => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   // ReactDOM.unmountComponentAtNode(div);
//   // setTimeout(() => done());
// });

// jest.mock('../../helpers/apiCalls/apiCalls');

// describe('App', () => {
//   window.location = { search: 'localHost.com' };
//   console.log(window.location);

//   let wrapper;
//   let mockStore;

//   beforeEach(() => {
//     wrapper = shallow(<App currentUser={store.currentUser} />);
//   });

//   it('should matchsnapshot', () => {
//     wrapper = mount(<App currentUser={store.currentUser} />);

//     expect(wrapper).toMatchSnapshot();
//   });
// });

// it('should render NavBar and Welcome and ContentRoute if state of code exists', () => {
//   wrapper.setState({ code: 'adfad' });
//   expect(wrapper).toMatchSnapshot();
// });

// it('should map the store correctly', () => {
//   const mockStore = {
//     ...store.currentUser
//   };
//   const mapped = mapStateToProps(mockStore);
//   expect(mapped.currentUser).toEqual(mockStore.currentUser);
// });

// it('should dispatch fn when using matchDispatchToProps function', () => {
//   const mockDispatch = jest.fn();
//   const mapped = mapDispatchToProps(mockDispatch);
//   mapped.setAccessToken();
//   expect(mockDispatch).toHaveBeenCalled();
// });
