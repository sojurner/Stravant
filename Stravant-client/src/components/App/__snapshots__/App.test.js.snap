// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should matchsnapshot 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <span
      className="header"
    >
      <h1
        className="welcome-title"
      >
        S†ra√an†
      </h1>
      <i
        className="fas fa-sign-in-alt"
        onClick={[Function]}
      />
    </span>
    <span
      className="logos"
    >
      <div
        className="spinning-globe"
      >
        Credits to Nick
      </div>
      <img
        className="connect-logo"
        height="75"
        onClick={[Function]}
        src="connect-logo.png"
        width="250"
      />
    </span>
    <div
      className="required-logo"
    >
      <img
        className="strava-powered"
        height="40"
        src="powered-by-strava.png"
        width="150"
      />
      <i
        class="fab fa-strava"
      />
      <img
        className="strava-powered"
        height="70"
        src="turing-logo.png"
        width="70"
      />
    </div>
  </div>
</BrowserRouter>
`;

exports[`App should render NavBar and Welcome and ContentRoute if state of code exists 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <span
      className="header"
    >
      <h1
        className="welcome-title"
      >
        S†ra√an†
      </h1>
      <i
        className="fas fa-sign-in-alt"
        onClick={[Function]}
      />
    </span>
    <Component />
    <Connect(Welcome) />
    <ContentRoute />
    <span
      className="logos"
    >
      <div
        className="spinning-globe"
      >
        Credits to Nick
      </div>
      <img
        className="connect-logo"
        height="75"
        onClick={[Function]}
        src="connect-logo.png"
        width="250"
      />
    </span>
    <div
      className="required-logo"
    >
      <img
        className="strava-powered"
        height="40"
        src="powered-by-strava.png"
        width="150"
      />
      <i
        class="fab fa-strava"
      />
      <img
        className="strava-powered"
        height="70"
        src="turing-logo.png"
        width="70"
      />
    </div>
  </div>
</BrowserRouter>
`;
